torchgen/native_function_generation.py

```diff
@@ -387,6 +388,9 @@ def add_generated_native_functions(
             # on CompositeImplicitAutograd operators (since we let them decompose).
             are_composite_implicit = all(
                 f.has_composite_implicit_autograd_kernel for f in d.values()
+            ) or all(
+                f.has_composite_implicit_autograd_nested_tensor_kernel
+                for f in d.values()
```

Comment:
I think this could actually break stuff - we should just remove this line. (too bad nothing failed in CI to tell you that, probably because it didnt' affect any existing operators today).

The idea here is that all operators in native_functions.yaml (that do **not** have `CompositeImplicitAutograd` kernels) are now required to have functional + out= variants.

It's now technically possible to have an op with a `CompositeImplicitAutogradNestedTensor` kernel, but not a `CompositeImplicitAutograd` kernel. We would still want the out= and functional variants to be generated for that operator.